src/DataViews/FunctionsDataView.cpp

```diff
@@ -248,76 +244,50 @@ void FunctionsDataView::OnContextMenu(const std::string& action, int menu_index,
-
-    bool match = true;
-
-    for (std::string& filter_token : m_FilterTokens) {
-      if (name.find(filter_token) == std::string::npos) {
-        match = false;
-        break;
-      }
-    }
+  const size_t number_of_threads = thread_pool_->GetPoolSize();
+  const size_t number_of_functions_per_thread = functions_.size() / number_of_threads;
 
-    if (match) {
-      indices.push_back(i);
-    }
-  }
+  std::vector<orbit_base::Future<std::vector<uint64_t>>> filtered_indices_per_thread;
+  filtered_indices_per_thread.reserve(number_of_threads);
 
-  indices_ = std::move(indices);
-#endif
-}
+  for (size_t thread = 0; thread < number_of_threads; ++thread) {
+    size_t begin = thread * number_of_functions_per_thread;
 
-void FunctionsDataView::ParallelFilter() {
-#ifdef _WIN32
-  const std::vector<const FunctionInfo*>& functions(functions_);
-  const auto prio = oqpi::task_priority::normal;
-  auto numWorkers = oqpi::default_helpers::scheduler().workersCount(prio);
-  // int numWorkers = oqpi::thread::hardware_concurrency();
-  std::vector<std::vector<int>> indicesArray;
-  indicesArray.resize(numWorkers);
-
-  oqpi::default_helpers::parallel_for(
-      "FunctionsDataViewParallelFor", functions.size(),
-      [&](int32_t a_BlockIndex, int32_t a_ElementIndex) {
-        std::vector<int>& result = indicesArray[a_BlockIndex];
-        const std::string& name = absl::AsciiStrToLower(
-            orbit_client_data::function_utils::GetDisplayName(*functions[a_ElementIndex]));
-        const std::string& module =
-            orbit_client_data::function_utils::GetLoadedModuleName(*functions[a_ElementIndex]);
-
-        for (std::string& filterToken : m_FilterTokens) {
-          if (name.find(filterToken) == std::string::npos &&
-              module.find(filterToken) == std::string::npos) {
-            return;
-          }
-        }
+    size_t end = (thread + 1) * number_of_functions_per_thread;
+    if (thread + 1 == number_of_threads) end = functions_.size();
 
-        result.push_back(a_ElementIndex);
-      });
+    if (end == begin) continue;
 
-  std::set<int> indicesSet;
-  for (std::vector<int>& results : indicesArray) {
-    for (int index : results) {
-      indicesSet.insert(index);
-    }
+    filtered_indices_per_thread.emplace_back(thread_pool_->Schedule([begin, end, this]() {
+      std::vector<uint64_t> indices;
+
+      for (size_t index = begin; index < end; ++index) {
+        const FunctionInfo* function = functions_[index];
+        std::string name =
+            absl::AsciiStrToLower(orbit_client_data::function_utils::GetDisplayName(*function)) +
+            orbit_client_data::function_utils::GetLoadedModuleName(*function);
+
+        const auto is_token_found = [&name](const std::string& token) {
+          return name.find(token) != std::string::npos;
+        };
+
+        if (std::all_of(m_FilterTokens.begin(), m_FilterTokens.end(), is_token_found)) {
```

Comment:
This looks like a good chance to rename `m_FilterTokens` to `filter_tokens_` :)

Reply:
Done.